legacy/reqcounter/reqcounter.go

```diff
@@ -25,70 +25,123 @@ import (
-	since    time.Time  // When the current counter began recording requests.
+	Mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	Requests uint64        // Number of HTTP requests since recording started.
+	Since    time.Time     // When the current counter began recording requests.
+	Interval time.Duration // The duration of time between each log.
 }
 
-// log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+// increment adds 1 to the request counter, signifying another call to GCR.
+func (rc *RequestCounter) Increment() {
+	rc.Mutex.Lock()
+	rc.Requests++
+	rc.Mutex.Unlock()
+}
+
+// Flush records the number of HTTP requests found and resets the counter.
+func (rc *RequestCounter) Flush() {
 	// Hold onto the lock when reading & writing the counter.
-	rc.mutex.Lock()
-	defer rc.mutex.Unlock()
+	rc.Mutex.Lock()
+	defer rc.Mutex.Unlock()
 
 	// Log the number of requests within this measurement window.
-	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("2006-01-02 15:04:05"), rc.requests)
-	logrus.Debug(msg)
+	msg := fmt.Sprintf("From %s to %s [%d min] there have been %d requests to GCR.", rc.Since.Format(timestampFormat), time.Now().Format(timestampFormat), rc.Interval/time.Minute, rc.Requests)
+	Debug(msg)
 
 	// Reset the counter.
-	rc.requests = 0
-	rc.since = time.Now()
+	rc.reset()
+}
+
+// reset clears the request counter and stamps the current time of reset.
+// TODO: @tylerferrara in the future, use the request counter field 'persistent' to check
+// if the counter should be reset.
+func (rc *RequestCounter) reset() {
+	rc.Requests = 0
+	rc.Since = time.Now()
+}
+
+// watch continuously logs the counter at the specified intervals.
+func (rc *RequestCounter) watch() {
+	go func() {
+		for {
+			Sleep(rc.Interval)
+			rc.Flush()
+		}
+	}()
+}
+
+// NetworkMonitor is the primary means of monitoring network traffic between CIP and GCR.
+type NetworkMonitor struct {
+	Counters []*RequestCounter
+}
+
+// increment adds 1 to each request counter, signifying a new request has been made to GCR.
+func (nm *NetworkMonitor) increment() {
+	for _, rc := range nm.Counters {
+		rc.Increment()
+	}
+}
+
+// Log begins logging each counter at their specified intervals.
+func (nm *NetworkMonitor) Log() {
+	for _, rc := range nm.Counters {
+		rc.watch()
+	}
 }
 
 const (
-	// measurementWindow specifies the length of time to wait before logging the RequestCounter. Since Google's
+	// MeasurementWindow specifies the length of time to wait before logging the RequestCounter. Since Google's
 	// Container Registry specifies a quota of 50,000 HTTP requests per 10 min, the window
 	// for recording requests is set to 10 min.
+	// NOTE: This metric is only a rough approximation of the actual GCR quota. The specific 10min measurement
+	// is ambiguous, and therefore difficult to mimic correctly.
```

Comment:
Please explain why it's ambiguous, so that the reader does not have to guess for themselves.

Reply:
ack.